barretenberg/cpp/src/barretenberg/bb/main.cpp

```diff
@@ -1,1595 +1,741 @@
-    using Prover = UltraProver_<Flavor>;
-
-    uint32_t honk_recursion = 0;
-    if constexpr (IsAnyOf<Flavor, UltraFlavor, UltraKeccakFlavor, UltraKeccakZKFlavor>) {
-        honk_recursion = 1;
-    } else if constexpr (IsAnyOf<Flavor, UltraRollupFlavor>) {
-        honk_recursion = 2;
-    }
-
-    // TODO(https://github.com/AztecProtocol/barretenberg/issues/1180): Don't init grumpkin crs when unnecessary.
-    init_grumpkin_crs(1 << CONST_ECCVM_LOG_N);
-
-    // Lambda function to ensure the builder gets freed before generating the vk. Vk generation requires initialing the
-    // pippenger runtime state which leads to it being the peak, when its functionality is purely for debugging purposes
-    // here.
-    auto prover = [&] {
-        const acir_format::ProgramMetadata metadata{ .recursive = recursive, .honk_recursion = honk_recursion };
-        acir_format::AcirProgram program{ get_constraint_system(bytecodePath, metadata.honk_recursion) };
-        if (!witnessPath.empty()) {
-            program.witness = get_witness(witnessPath);
+    /***************************************************************************************************************
+     * Subcommand: prove_tube
+     ***************************************************************************************************************/
+    CLI ::App* prove_tube_command = app.add_subcommand("prove_tube", "");
+    prove_tube_command->group(""); // hide from list of subcommands
+    add_verbose_flag(prove_tube_command);
+    add_debug_flag(prove_tube_command);
+    add_crs_path_option(prove_tube_command);
+    std::string prove_tube_output_path{ "./target" };
+    add_output_path_option(prove_tube_command, prove_tube_output_path);
+
+    /***************************************************************************************************************
+     * Subcommand: verify_tube
+     ***************************************************************************************************************/
+    CLI::App* verify_tube_command = app.add_subcommand("verify_tube", "");
+    verify_tube_command->group(""); // hide from list of subcommands
+    add_verbose_flag(verify_tube_command);
+    add_debug_flag(verify_tube_command);
+    add_crs_path_option(verify_tube_command);
+    // doesn't make sense that this is set by -o but that's how it was
+    std::string tube_proof_and_vk_path{ "./target" };
+    add_output_path_option(verify_tube_command, tube_proof_and_vk_path);
+
+    /***************************************************************************************************************
+     * Build the CLI11 App
+     ***************************************************************************************************************/
+
+    CLI11_PARSE(app, argc, argv);
+    print_active_subcommands(app);
+    vinfo(flags);
+
+    debug_logging = flags.debug;
+    verbose_logging = debug_logging || flags.verbose;
+
+    // prob this construction is too much
```

Comment:
It might be possible to avoid this chain of if-elses by using callbacks on each subcommand. At least that's what I would've expected and seems supported: https://github.com/CLIUtils/CLI11?tab=readme-ov-file#callbacks

(but of course I only skimmed through it)